logpush-to-bigquery/index.js

```diff
@@ -26,17 +22,68 @@ async function gcsbq (file, context) {
     ignoreUnknownValues: true
   }
 
-  const addToTable = async (tableId) => {
+  const addToTable = async tableId => {
     const dataset = await bigquery.dataset(datasetId).get({ autoCreate: true })
     const table = await dataset[0].table(tableId).get({ autoCreate: true })
-    return table[0].load(filename, metadata)
+    return table[0].createLoadJob(files, metadata)
   }
 
   try {
-    await addToTable(tableId)
+    return addToTable(tableId)
   } catch (e) {
     console.log(e)
   }
 }
 
-module.exports.gcsbq = gcsbq
+async function writeLog (logData) {
+  logData = logData.reduce((acc, current) => [...acc, current.name], [])
+
+  const metadata = {
+    resource: { type: 'global' },
+    severity: 'INFO'
+  }
+
+  const entry = log.entry(metadata, logData)
+  await log.write(entry)
+  console.log(
+    `Loaded to ${process.env.DATASET}.${process.env.TABLE}: ${logData}`
+  )
+}
+
+module.exports.runLoadJob = async function (message, context) {
+  if (!context) {
+    context = {}
+    context.timestamp = new Date().toISOString()
+  }
+  context.timestamp = DateTime.fromISO(context.timestamp)
+
+  const loadJobDeadline = context.timestamp
+    .setZone('GMT')
+    .minus({ minutes: 15 })
+    .startOf('minute')
+
+  const [deadlineDate, deadlineDt] = [
+    loadJobDeadline.toFormat('yyyyMMdd'),
+    loadJobDeadline.toFormat(`yyyyMMdd'T'hhmm`)
+  ]
+
+  let stackdriverEntry = []
+
+  try {
+    let logFiles = await bucket.getFiles({
+      autoPaginate: false,
+      maxResults: 5000,
+      prefix: `${process.env.DIRECTORY}${deadlineDate}/${deadlineDt}`
```

Comment:
Per discussion with Jamie, we're going to use the log start time and increase the delay on the cron job